Compiled from "CWE396_Catch_Generic_Exception__Throwable_07.java"
public class testcases.CWE396_Catch_Generic_Exception.CWE396_Catch_Generic_Exception__Throwable_07 extends testcasesupport.AbstractTestCase {
  private int privateFive;

  public testcases.CWE396_Catch_Generic_Exception.CWE396_Catch_Generic_Exception__Throwable_07();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method testcasesupport/AbstractTestCase."<init>":()V
       4: aload_0
       5: iconst_5
       6: putfield      #7                  // Field privateFive:I
       9: return

  public void bad() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #7                  // Field privateFive:I
       4: iconst_5
       5: if_icmpne     25
       8: ldc           #13                 // String Test
      10: invokestatic  #15                 // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      13: pop
      14: goto          25
      17: astore_1
      18: ldc           #23                 // String Caught Throwable
      20: invokestatic  #25                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      23: aload_1
      24: athrow
      25: return
    Exception table:
       from    to  target type
           8    14    17   Class java/lang/Throwable

  private void good1() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #7                  // Field privateFive:I
       4: iconst_5
       5: if_icmpeq     16
       8: ldc           #31                 // String Benign, fixed string
      10: invokestatic  #25                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      13: goto          33
      16: ldc           #13                 // String Test
      18: invokestatic  #15                 // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      21: pop
      22: goto          33
      25: astore_1
      26: ldc           #35                 // String Caught Exception
      28: invokestatic  #25                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      31: aload_1
      32: athrow
      33: return
    Exception table:
       from    to  target type
          16    22    25   Class java/lang/NumberFormatException

  private void good2() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #7                  // Field privateFive:I
       4: iconst_5
       5: if_icmpne     25
       8: ldc           #13                 // String Test
      10: invokestatic  #15                 // Method java/lang/Integer.parseInt:(Ljava/lang/String;)I
      13: pop
      14: goto          25
      17: astore_1
      18: ldc           #35                 // String Caught Exception
      20: invokestatic  #25                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      23: aload_1
      24: athrow
      25: return
    Exception table:
       from    to  target type
           8    14    17   Class java/lang/NumberFormatException

  public void good() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: invokevirtual #37                 // Method good1:()V
       4: aload_0
       5: invokevirtual #40                 // Method good2:()V
       8: return

  public static void main(java.lang.String[]) throws java.lang.ClassNotFoundException, java.lang.InstantiationException, java.lang.IllegalAccessException;
    Code:
       0: aload_0
       1: invokestatic  #43                 // Method mainFromParent:([Ljava/lang/String;)V
       4: return
}
